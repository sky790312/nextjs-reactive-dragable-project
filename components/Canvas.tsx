import React from 'react'
import styled from 'styled-components'
import Block from '@/components/Block'
import { useSelector, shallowEqual } from 'react-redux'
import { RootState } from '@/store/index'
import { useRouter } from 'next/router'

const Canvas: React.FC = React.memo(() => {
  const router = useRouter()
  const blocks = useSelector(
    (state: RootState) => state.block.blocks,
    shallowEqual,
  )
  const pageBlocks = blocks.filter(
    block => block.name.split(' ')[1].charAt(0) === router?.query?.id,
  )

  return (
    <CanvasWrapper>
      {pageBlocks.map(block => (
        <Block key={block.id} block={block} />
      ))}
    </CanvasWrapper>
  )
})

export default Canvas

const CanvasWrapper = styled.div`
  position: relative;
  background: white;
  overflow: hidden;
`
