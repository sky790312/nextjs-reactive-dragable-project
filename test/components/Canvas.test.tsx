import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import Canvas from '@/components/Canvas'

describe('Canvas component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Canvas />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
