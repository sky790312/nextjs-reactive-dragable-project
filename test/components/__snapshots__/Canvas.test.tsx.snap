// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Canvas__CanvasWrapper-sc-5gmsqo-0 hInTKC"
  />
</DocumentFragment>
`;
